lib/src/server.ml

```diff
@@ -223,35 +231,110 @@ let run config host port root trim_period trim_size =
         | Some h -> h
       in
       let block_get =
-        let f t h = Blocks.get ~f:Core.In_channel.read_all t (hash h) in
-        Async_rpc.implement Rpc.block_get f
+        let f { t; _ } h =
+          let* () = Logs_async.info (fun m -> m "GET blocks/%s" h) in
+          let block =
+            let f path =
+              let* reader = Async.Reader.open_file path in
+              Async.return @@ Async.Reader.pipe reader
+            in
+            Blocks.get ~f t (hash h)
```

Comment:
I found it difficult to understand what `hash h` does, so I had to look up the definition; `digest_of_hex_exn` would be a more self-documenting name.